### PR TITLE
enhancement: Refactor NewLogger with options builder pattern

### LINKED ISSUE #1355: Refactor logger.NewLogger with options builder pattern
We should refactor the [logger.NewLogger](https://pkg.go.dev/github.com/openfga/openfga@v1.4.3/pkg/logger#NewLogger) signature to use the options builder pattern and if an option is not provided we just default it.

For example,
```go
NewLogger(
  WithFormat("text"),
  WithLevel("info"),
  WithTimestampFormat("ISO8601"),
)
```

_Originally posted by @jon-whit in https://github.com/openfga/openfga/pull/1330#discussion_r1484561151_

### PR DESCRIPTION
**Pull request by lekaf974**
## Description

Fix #1355 

## References

Commented in #1330

### CODE REVIEW

File: pkg/logger/logger.go [hunk 1]
```diff
@@ -95,14 +94,51 @@ func (l *ZapLogger) FatalWithContext(ctx context.Context, msg string, fields ...
 	l.Logger.Fatal(msg, fields...)
 }
 
-func NewLogger(logFormat, logLevel, logTimestampFormat string) (*ZapLogger, error) {
-	if logLevel == "none" {
+// OptionsLogger Implement options for logger
```
Comment (adriantam) [inline pkg/logger/logger.go:97]:
```suggestion
// OptionsLogger implements options for logger
```

File: pkg/logger/logger.go [hunk 2]
```diff
@@ -95,14 +94,51 @@ func (l *ZapLogger) FatalWithContext(ctx context.Context, msg string, fields ...
...
+type OptionsLogger struct {
+	Format          string
+	Level           string
+	TimestampFormat string
```
Comment (adriantam) [inline pkg/logger/logger.go:101]:
```suggestion
	format          string
	level           string
	timestampFormat string
```

The member variable should be private.  Changes to these member variables should be done via WithX methods below.

File: pkg/logger/logger.go (hunk 1 above)
Comment (lekaf974) [reply pkg/logger/logger.go]: Fixed

File: pkg/logger/logger.go (hunk 2 above)
Comment (lekaf974) [reply pkg/logger/logger.go]: Set private

File: pkg/logger/logger.go [hunk 3]
```diff
@@ -3,7 +3,6 @@ package logger
 import (
 	"context"
 	"fmt"
-
```
[the remark this answers is not in the document]
Comment (lekaf974) [reply pkg/logger/logger.go]: done

File: pkg/logger/logger.go [hunk 4]
```diff
@@ -95,14 +95,51 @@ func (l *ZapLogger) FatalWithContext(ctx context.Context, msg string, fields ...
 	l.Logger.Fatal(msg, fields...)
 }
 
-func NewLogger(logFormat, logLevel, logTimestampFormat string) (*ZapLogger, error) {
-	if logLevel == "none" {
+// OptionsLogger Implements options for logger
+type OptionsLogger struct {
+	format          string
+	level           string
+	TimestampFormat string
```
Comment (adriantam) [inline pkg/logger/logger.go:102]: TImestampFormat is still public

Comment (lekaf974) [reply pkg/logger/logger.go]: fixed

**Review by adriantam — APPROVED** [review]

**Pull request merged**